pkg/notifications/notification.go

```diff
@@ -0,0 +1,70 @@
+// SPDX-FileCopyrightText: 2024 Greenbone AG <https://greenbone.net>
+//
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+// Package Notifications provides a client to communicate with the OpenSight Notification Service [github.com/greenbone/opensight-notification-service]
+package notifications
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+)
+
+const basePath = "/api/notification-service"
+const createNotificationPath = "/notifications"
+
+// Client can be used to send notifications
+type Client struct {
+	httpClient                 *http.Client
+	notificationServiceAddress string
+}
+
+// NewClient returns a new [Client] with the notification service address (host:port) set.
+func NewClient(httpClient *http.Client, notificationServiceAddress string) *Client {
+	return &Client{
+		httpClient:                 httpClient,
+		notificationServiceAddress: notificationServiceAddress,
+	}
+}
+
+// CreateNotification sends a notification to the notification service
+func (c *Client) CreateNotification(notification Notification) error {
+	var notificationSerialized bytes.Buffer
+	err := json.NewEncoder(&notificationSerialized).Encode(notification)
+	if err != nil {
+		return fmt.Errorf("failed to serialize notification object: %w", err)
+	}
+
+	// create request
+	createNotificationEndpoint, err := url.JoinPath(c.notificationServiceAddress, basePath, createNotificationPath)
+	if err != nil {
+		return fmt.Errorf("invalid url '%s': %w", c.notificationServiceAddress, err)
+	}
+	request, err := http.NewRequest(http.MethodPost, createNotificationEndpoint, &notificationSerialized)
+	if err != nil {
+		return fmt.Errorf("failed to build request: %w", err)
+	}
+
+	// execute request
+	response, err := c.httpClient.Do(request)
+	if err != nil {
+		return fmt.Errorf("failed to send request to %s: %w", createNotificationEndpoint, err)
```

Comment:
You could think about using errors that you can distinguish by type like https://github.com/greenbone/asset-management-backend/blob/main/internal/entities/errors/errors.go or https://github.com/greenbone/asset-management-backend/blob/main/internal/entities/errors/applianceError.go (see https://github.com/greenbone/asset-management-backend/blob/main/internal/entities/errors/ and https://github.com/greenbone/asset-management-backend/blob/main/internal/controllers/ginController/ginErrorHandler/errorHandler.go for some context).

But this is an add-on, which could be done later on as well. Also this would raise the question in we should move https://github.com/greenbone/asset-management-backend/blob/main/internal/entities/errors/UserError.go to the library first (which we intended to do at some point anyways)

Reply:
I think it definitely makes sense to have some custom error types like not-found etc. in this library which can be reused across the services.

However for the notification service I currently don't see the use case for a non-binary error handling.